tools/toolchains/arm.py

```diff
@@ -71,7 +77,17 @@ def __init__(self, target, notify=None, macros=None,
             raise NotSupportedException(
                 "this compiler does not support the core %s" % target.core)
 
-        if getattr(target, "default_toolchain", "ARM") == "uARM":
+        if  getattr(target, "default_lib", "std") == "small":
+            if (
+                hasattr(target, "supported_c_libs") == False
+                or "arm" not in target.supported_c_libs
+                or "small" not in target.supported_c_libs["arm"]
+            ):
+                raise NotSupportedException(
+                  "target.default_lib small not supported for this target."
```

Comment:
Write "small C library not supported for this target" so you don't have to change the message if/when default_lib becomes "c_lib".

Reply:
"small C library *option* not supported for this target".